### PR TITLE
Added support for HuggingFace model download

### PR DESCRIPTION
**Pull request by anandhu-eng**
Added support to download the hugging face model based on the inbuilt variations or as variation based on user defined input(model name as in Hugging face hub).
Script folder name: get-ml-model-huggingface-zoo
Inbuilt variation: pierreguillou_bert_base_cased_squad_v1.1_portuguese
Hugging face bug link fro particular model: https://huggingface.co/pierreguillou/bert-base-cased-squad-v1.1-portuguese

### CODE REVIEW

File: cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "deepsparse",
+    "model-zoo",
+    "sparse-zoo",
+    "huggingface"
+  ],
+  "uid": "53cf8252a443446a",
+  "variations": {
+    "model-stub.#": {
+      "env": {
+        "CM_MODEL_ZOO_STUB": "#"
```
Comment (arjunsuresh) [inline cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json:38]: This looks nice. We can add some example stubs in "README-about.md"

File: cm-mlops/script/get-ml-model-huggingface-zoo/download_sparse.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from setfit import SetFitModel
+import os
+
+model_stub= os.environ.get('CM_MODEL_ZOO_STUB', '')
+curr_script_path = os.environ.get('CM_TMP_CURRENT_SCRIPT_PATH','')
```
[the remark this answers is not in the document]
Comment (anandhu-eng) [reply cm-mlops/script/get-ml-model-huggingface-zoo/download_sparse.py:5]: Have modified the script accordingly 👍🏼

File: cm-mlops/script/get-ml-model-huggingface-zoo/run.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+${CM_PYTHON_BIN_WITH_PATH} ${CM_TMP_CURRENT_SCRIPT_PATH}/download_sparse.py
```
[the remark this answers is not in the document]
Comment (anandhu-eng) [reply cm-mlops/script/get-ml-model-huggingface-zoo/run.sh:2]: modified 👍🏼

File: cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json (hunk 1 above)
Comment (anandhu-eng) [reply cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json:38]: Sure. Have added example stubs from huggingface having download count of greater than 1k

File: cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "new_env_keys": [
+    "CM_ML_MODEL*",
+    "CM_MODEL_ZOO_STUB"
+  ],
+  "tags": [
+    "get",
+    "ml-model",
+    "model",
+    "zoo",
+    "deepsparse",
```
[the remark this answers is not in the document]
Comment (anandhu-eng) [reply cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json:29]: removed 👍🏼

File: cm-mlops/script/get-ml-model-huggingface-zoo/download_sparse.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+print(f"Downloading model {model_stub}")
+stub = f"{model_stub}"
+model = SetFitModel.from_pretrained(stub, cache_dir='.')
+
+model._save_pretrained(stub)
+
+with open('tmp-run-env.out', 'w') as f:
+    f.write(f"CM_ML_MODEL_FILE_WITH_PATH={curr_script_path+'/'+stub}")
```
[the remark this answers is not in the document]
Comment (anandhu-eng) [reply cm-mlops/script/get-ml-model-huggingface-zoo/download_sparse.py:13]: Modified 👍🏼

File: cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  "deps": [
+    {
+      "names": [
+        "python3",
+        "python"
+      ],
+      "tags": "get,python3"
+    },
+    {
+      "tags": "get,generic-python-lib,_sparsezoo"
```
[the remark this answers is not in the document]
Comment (arjunsuresh) [reply cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json]: That's great. If we don't need any python package, we can just remove that tags entry completely.

File: cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json (hunk 1 above)
Comment (arjunsuresh) [reply cm-mlops/script/get-ml-model-huggingface-zoo/_cm.json:36]: Thank you @anandhu-eng . That's useful.

File: cm-mlops/script/get-ml-model-huggingface-zoo/download_model.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+model_stub= os.environ.get('CM_MODEL_ZOO_STUB', '')
+print(f"Downloading model {model_stub}")
+stub = f"{model_stub}"
+model = SetFitModel.from_pretrained(stub, cache_dir=os.getcwd())
+
+model._save_pretrained(stub)
+
+with open('tmp-run-env.out', 'w') as f:
+    f.write(f"CM_ML_MODEL_FILE_WITH_PATH={os.path.join(os.getcwd(),'/',stub)}")
```
Comment (arjunsuresh) [inline cm-mlops/script/get-ml-model-huggingface-zoo/download_model.py:12]: It should be os.path.join(os.getcwd(), stub)

Comment (anandhu-eng) [reply cm-mlops/script/get-ml-model-huggingface-zoo/download_model.py]: modified 👍🏼

**Review by arjunsuresh — APPROVED** [review]
LGTM. Thank you Anandhu for a very useful contribution!

**Pull request merged**